policy_maker/app.py

```diff
@@ -100,6 +100,22 @@ def create_policy_document(bucket_name, principals):
           's3:*MultipartUpload*'
         ],
         'Resource': bucket_objects_arn
+      },
+      {
+        'Sid': 'RequireCanonicalId',
+        'Effect': 'Deny',
+        'Principal': {
+          'AWS': principals
```

Comment:
Can you confirm that the `principals` would correspond to the users listed in `S3UserARNs` in the SC product?

Reply:
As you can see, just above, the `principals` are set the exact same way for WriteAccess and ReadAccess.  this is not new code @BrunoGrandePhD the service catalog is running this code in production and when we look at the policies for the buckets provisioned by the SC we see that `principals` do indeed correspond to S3UserARNs for ReadAccess and WriteAccess.

Reply:
Thanks for confirming! I just didn't want to assume one way or another, nor spend an entire afternoon figuring out how the various repos work together.

Reply:
@BrunoGrandePhD I think this logic should be changed.  We should _not_ deny `PutObject` access to _all_ in the 'principals' list if they don't add the `bucket-owner-full-control` ACL.  Instead we should only apply this condition to principals who are **not** in the account that owns the bucket.  I.e., if a principal in the account that owns the bucket uploads a file without the ACL that is OK because the bucket owner *already* owns the uploaded object.    This change is important because, as you discovered for us, when uploading through the S3 console there is no way to add an ACL to the created object.